### PR TITLE
PostCSS setup

### PR DESCRIPTION
**Pull request by kpuputti**
This PR sets up the configuration to use [PostCSS](http://postcss.org/) in the new Frontend architecture, i.e. the React components within the `client/` directory.

The idea is to use [cssnext](http://cssnext.io/) for stylesheet preprocessing and basic functionality, with variables coming from JavaScript through the Webpack config. In addition to cssnext, at least mixins (and possibly functions) are added with a PostCSS plugin. [cssnano](http://cssnano.co/) is used for minification and compression, and [stylelint](http://stylelint.io/) for linting.

TODO:

 - [x] Setup cssnext (with variables coming from the Webpack config)
 - [x] Setup mixins
 - [x] Convert SCSS styles (within `client/`) to use the new setup
 - [x] Split PostCSS output from `application.scss`
 - [x] ~~Ensure sourcemaps are working~~ (working in Styleguide)
 - [x] ~~Setup stylelint and linting configuration~~ (postpone to another PR)
 - [x] Setup cssnano (through css-loader)
 - [x] Git history cleanup
 - [x] Fix duplicate plugin warnings
 - [x] Add CSS vars to Styleguide

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -6,46 +6,46 @@
...
+    "build:production:client": "NODE_ENV=production webpack --config webpack.client.config.js",
+    "build:production:server": "NODE_ENV=production webpack --config webpack.server.config.js",
+    "build:dev:client": "webpack -w --config webpack.client.config.js",
+    "build:dev:server": "webpack -w --config webpack.server.config.js",
     "print-phantomjs-version": "phantomjs --version",
     "start-phantomjs": "phantomjs --webdriver=8910",
     "styleguide": "start-storybook -p 9001"
   },
   "dependencies": {
-    "autoprefixer": "6.3.6",
```
Comment (mporkola) [inline client/package.json]: what happened to autoprefixer?

Comment (kpuputti) [inline client/package.json]: cssnext includes it and warns about a duplicate if it is added separately. The same autoprefixer config is given to cssnext.

Comment (mporkola) [inline client/package.json]: thought it must be included somewhere 👍

**Pull request merged**